### PR TITLE
Fix code gen for table model deserializer

### LINKED ISSUE #67: Incomplete deserializer code generated for !table model
### Expected behaviour

The generated Deserializer class should have deserialize<TableName>Table methods for all !table types

### Actual behaviour
A class is generated with missing deserialize<TableName>Table methods. As a result the generated java code cannot be compiled.

### Steps to reproduce the behaviour

Example model
` cat relationalmodel.sysl`

```
UserModel [package="io.sysl.reljam.gen.relational"]:
    !table User:
        UserID <: int [~pk, ~autoinc]
        FirstName <: string(50)
        LastName <: string(50)
        PrimaryAddressID <: Address.AddressID

    !table Address:
        AddressID <: int [~pk, ~autoinc]
        Line1 <: string(100)
        Line2 <: string(100)
        City <: string(100)
        State <: string(100)
        ZipCode <: string(10)
        Country <: string(100)

    !table UserAddressRel:
        UserID <: User.UserID
        AddressID <: Address.AddressID
        AddressType <: string(10)

UserFacade [package="com.test.example.relational.facade"]:
    !wrap UserModel:
        !type User
        !type Address
        !type UserAddressRel
```

`reljam --out gen/ model /relationalmodel UserModel`

Some of the methods are missing in the generated output.

See `gen/io/sysl/reljam/gen/relational/UserModelJsonDeserializer.java`

### PR DESCRIPTION
**Pull request by sahejsingh**
Fixes #67 

Changes proposed in this pull request:
- Code was not getting generated for table based model's. `continue` was in the wrong place.
- Added tests for test serialisation and deserialisation of !table based models.

@anz-bank/sysl-developers

### CODE REVIEW

File: test/java/src/test/resources/test4.xml [hunk 1]
```diff
@@ -0,0 +1 @@
+<UserModel><AddressList><Address><AddressID>1</AddressID><Address1>3000 Flinders Street</Address1><City>Melbourne</City><Country>AU</Country><State>VIC</State><ZipCode>3001</ZipCode></Address><Address><AddressID>3</AddressID><Address1>ANZ Building</Address1><Address2>833 Collins Street</Address2><City>Melbourne</City><Country>AU</Country><State>VIC</State><ZipCode>3000</ZipCode></Address></AddressList><UserAddressRelList><UserAddressRel><AddressID>1</AddressID><UserID>2</UserID><AddressType>Home</AddressType></UserAddressRel><UserAddressRel><AddressID>3</AddressID><UserID>2</UserID><AddressType>Office</AddressType></UserAddressRel></UserAddressRelList><UsersList><Users><UserID>2</UserID><PrimaryAddressID>1</PrimaryAddressID><FirstName>John</FirstName><LastName>Smith</LastName></Users></UsersList></UserModel>
```
Comment (juliaogris) [inline test/java/src/test/resources/test4.xml]: Could you please format this with newlines and indents?

File: test/java/build.gradle [hunk 2]
```diff
@@ -24,7 +24,31 @@ task generateSysl(type:Exec) {
   }
 }
 
-compileJava.dependsOn generateSysl
+task generateUserModel(type:Exec) {
+  commandLine "reljam",  "--out", "$tempDirectory", "model", "/relationalmodel", "UserModel"
+
```
Comment (juliaogris) [inline test/java/build.gradle]: Can you please delete all lines below until the closing brace of `generateUserModel`? That will make for a way terser code:
```
task generateUserModel(type:Exec) {
  commandLine "reljam",  "--out", "$tempDirectory", "model", "/relationalmodel", "UserModel"
}
```
I've tested it locally, gradle does not capture the output, you still get useful error messages without these lines.

Same for all other `task generate*(type:Exec)`

File: test/java/src/test/resources/test4.xml (hunk 1 above)
Comment (sahejsingh) [reply test/java/src/test/resources/test4.xml]: Unfortunately no. ATM Deserialiser does not like formatted input.

Comment (juliaogris) [reply test/java/src/test/resources/test4.xml]: I see, not to worry then.

**Comment by juliaogris** [comment]
Great work with the coverage increase! Thanks a lot!

File: .travis.yml [hunk 3]
```diff
@@ -8,6 +8,9 @@ matrix:
     after_success:
     - pip install codecov -e .
     - coverage run --source=src/sysl -m py.test
+    - coverage run -a --source=src/sysl $(which reljam) --root $(pwd) model /test/java/relationalmodel UserModel
+    - coverage run -a --source=src/sysl $(which reljam) --root $(pwd) facade /test/java/relationalmodel UserFacade
+    - coverage run -a --source=src/sysl $(which reljam) --root $(pwd) model /test/java/tuplecomplex UserFormComplex
```
Comment (juliaogris) [inline .travis.yml]: Hmm, you are not really running tests here, are you?

Comment (sahejsingh) [reply .travis.yml]: Ah. yea. I know. That was just an experiment to see if coverage increases if I run the tool directly. I will remove this.

Comment (juliaogris) [reply .travis.yml]: oh :( I got really excited about the increased coverage ;)

File: src/sysl/reljam/reljam.py [hunk 4]
```diff
@@ -212,3 +213,8 @@ def main():
     if entities:
         raise RuntimeError('Some entities not output as expected: ' +
                            ', '.join(sorted(entities)))
+
+
+if __name__ == '__main__':
+    main()
+    sys.exit(0)
```
Comment (marcelocantos) [inline src/sysl/reljam/reljam.py]: Redundant

File: test/java/src/test/resources/test4.xml (hunk 1 above)
Comment (marcelocantos) [reply test/java/src/test/resources/test4.xml]: Raise an issue to fix this.

File: src/sysl/reljam/reljam.py (hunk 4 above)
Comment (juliaogris) [reply src/sysl/reljam/reljam.py]: Actually, good point.  The `if __name__ == '__main__':` entry point is in `reljam/__main__.py`.  When you run `reljam` it with `python -m` only the package needs to be specified e.g. `python -m sysl.reljam`

Comment (marcelocantos) [reply src/sysl/reljam/reljam.py]: Is was referring only to the sys.exit call. I don't know about the surrounding context.

Comment (juliaogris) [reply src/sysl/reljam/reljam.py]: Ok.
See my comment as an additional one then.

**Comment by juliaogris** [comment]
Great job on the coverage increase 👍
With coverage 70% you get to close #44 😉

File: src/sysl/reljam/reljam.py [hunk 5]
```diff
@@ -212,3 +213,7 @@ def main():
     if entities:
         raise RuntimeError('Some entities not output as expected: ' +
                            ', '.join(sorted(entities)))
+
+
+if __name__ == '__main__':
```
Comment (juliaogris) [inline src/sysl/reljam/reljam.py]: Please remove this, it's already covered in __main__.py

File: test/java/src/test/resources/test4.xml (hunk 1 above)
Comment (juliaogris) [reply test/java/src/test/resources/test4.xml]: Yup, please raise issue and refer to it here.

Comment (sahejsingh) [reply test/java/src/test/resources/test4.xml]: Raised issue #81

**Pull request merged**